modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -411,6 +413,12 @@ public void setBaseline(Set<String> baselineNodes) {
         }
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public IgniteCompute compute() {
+        return new IgniteComputeImpl();
```

Comment:
I know that it's good enough for now, but why not field?

Reply:
It's still not clear how it will be created. To make it a little less worrying, I now changed it to lazy instantiation.